### PR TITLE
Change GetTrajectoryNodes() to MapById.

### PR DESCRIPTION
**Pull request by cschuet**

### CODE REVIEW

File: cartographer/mapping_2d/sparse_pose_graph_test.cc [hunk 1]
```diff
@@ -192,13 +197,14 @@ TEST_F(SparsePoseGraphTest, NoMovement) {
   MoveRelative(transform::Rigid2d::Identity());
   sparse_pose_graph_->RunFinalOptimization();
   const auto nodes = sparse_pose_graph_->GetTrajectoryNodes();
-  ASSERT_THAT(nodes.size(), ::testing::Eq(1u));
-  EXPECT_THAT(nodes[0].size(), ::testing::Eq(3u));
-  EXPECT_THAT(nodes[0][0].global_pose,
+  ASSERT_THAT(std::vector<int>{0},
+              ::testing::ContainerEq(ToVector(nodes.trajectory_ids())));
```
Comment (wohe) [inline cartographer/mapping_2d/sparse_pose_graph_test.cc]: Swap the vectors, i.e. actual value `nodes.trajectory_ids()` first, expected value in the matcher? Similar below.

File: cartographer/mapping_2d/sparse_pose_graph_test.cc [hunk 2]
```diff
@@ -173,6 +173,11 @@ class SparsePoseGraphTest : public ::testing::Test {
     MoveRelativeWithNoise(movement, transform::Rigid2d::Identity());
   }
 
+  template <typename T>
+  std::vector<int> ToVector(const T& t) {
```
Comment (wohe) [inline cartographer/mapping_2d/sparse_pose_graph_test.cc]: `T& t` -> `Range& range`?

Comment (wohe) [inline cartographer/mapping_2d/sparse_pose_graph_test.cc]: `ToVectorInt`?

File: cartographer/mapping/sparse_pose_graph.h [hunk 3]
```diff
@@ -114,7 +114,7 @@ class SparsePoseGraph {
   virtual transform::Rigid3d GetLocalToGlobalTransform(int trajectory_id) = 0;
 
   // Returns the current optimized trajectories.
-  virtual std::vector<std::vector<TrajectoryNode>> GetTrajectoryNodes() = 0;
+  virtual MapById<NodeId, TrajectoryNode> GetTrajectoryNodes() = 0;
```
Comment (wohe) [inline cartographer/mapping/sparse_pose_graph.h]: Also drop the `mapping::` at `mapping::MapById<mapping::SubmapId, ...`?

Comment (cschuet) [reply cartographer/mapping/sparse_pose_graph.h]: Done

File: cartographer/mapping_2d/sparse_pose_graph_test.cc (hunk 2 above)
Comment (cschuet) [reply cartographer/mapping_2d/sparse_pose_graph_test.cc]: Done

Comment (cschuet) [reply cartographer/mapping_2d/sparse_pose_graph_test.cc]: Done

File: cartographer/mapping_2d/sparse_pose_graph_test.cc (hunk 1 above)
Comment (cschuet) [reply cartographer/mapping_2d/sparse_pose_graph_test.cc]: Done

**Pull request merged**